### PR TITLE
server: add proxyprotocol to http

### PR DESCRIPTION
**Pull request by xhebox**
### What problem does this PR solve?

Issue Number: ref #245

Problem Summary: If `NLB` enabled with proxy-protocol, the HTTP port may be sent PPv2 headers. This PR added the support.

What is changed and how it works:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Notable changes

- [x] Has configuration change
- [ ] Has HTTP API interfaces change
- [ ] Has tiproxyctl change
- [ ] Other user behavior changes

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Add support of proxy-protocol for HTTP API port
```

### CODE REVIEW

File: pkg/proxy/proxyprotocol/listener.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+			c.proxy, _, err = ParseProxyV2(c.Conn)
+			if err != nil {
+				return 0, err
+			}
+		}
+		c.inited = true
+	}
+	if len(c.proxyHdr) > 0 {
+		n = copy(b, c.proxyHdr)
+		c.proxyHdr = c.proxyHdr[n:]
```
Comment (djshow832) [inline pkg/proxy/proxyprotocol/listener.go:66]: Why do you preserve the header? I assume the client(NLB) always sends the header in the first packet.

Comment (xhebox) [reply pkg/proxy/proxyprotocol/listener.go:66]: NLB will, but others will not.

File: pkg/proxy/proxyprotocol/listener.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+				return 0, nil
+			}
+			// it is proxy protocol
+			c.buf.Reset()
+			c.proxy, _, err = ParseProxyV2(c.Conn)
+			if err != nil {
+				return 0, err
+			}
+		}
+		// prefixes mismatched
```
Comment (djshow832) [inline pkg/proxy/proxyprotocol/listener.go:65]: Wrong comment. It may also match and parse successfully.

**Review by djshow832 — COMMENTED** [review]
Rest LGTM

**Review by djshow832 — APPROVED** [review]

**Pull request merged**